src/NuGet.Core/NuGet.Packaging/PackageExtraction/PackageHelper.cs

```diff
@@ -23,6 +23,13 @@ public static class PackageHelper
 
         private const string ExcludeExtension = ".nupkg.sha512";
 
+        public static bool IsAssembly(string path)
+        { 
+            return path.EndsWith(".dll", StringComparison.OrdinalIgnoreCase) || 
+                   path.EndsWith(".winmd", StringComparison.OrdinalIgnoreCase) || 
+                   path.EndsWith(".exe", StringComparison.OrdinalIgnoreCase); 
+        }
+
         public static bool IsManifest(string path)
         {
             return Path.GetExtension(path).Equals(PackagingCoreConstants.NuspecExtension, StringComparison.OrdinalIgnoreCase);
```

Comment:
No, if the same package is to work on Linux and Windows, we have to treat files as case insensitive on Linux even if it wouldn't normally do that.  It might have come from Windows with the wrong case.